### PR TITLE
Add proxy access lookup model

### LINKED ISSUE #160: Add proxy request integration
These are params we can send to symphony when we make the request.

We will need to make a call out to a proxy script on symphony to know if the person is in the appropriate symphony group to have proxy requests.

proxy

### PR DESCRIPTION
**Pull request by cbeer**
Fixes #160 

screen shot 2015-06-09 at 4 14 24 pm

### CODE REVIEW

File: app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -23,6 +24,12 @@ def current_request
 
   protected
 
+  def check_if_proxy_sponsor
+    return if params[:proxy].present? || !current_user.sponsor?
```
Comment (jkeck) [inline app/controllers/requests_controller.rb]: I'm receiving an error that traces back to this line when submitting a page request with a library ID.  (I guess that means we've been missing an important integration test about submitting w/ Library ID only).

In the case when a request is submitted w/ a Library ID only (or name+email) the `current_user` is still anonymous.  The user object that *I think* we're interested in is accessible through the `current_request`.

Comment (cbeer) [inline app/controllers/requests_controller.rb]: Can you help me replicate this? I tried a simple page with just a library id and just name / email and it worked for me.

File: app/views/requests/sponsor_request.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<h1 id="dialogTitle">Share with your proxy group?</h1>
```
Comment (jkeck) [inline app/views/requests/sponsor_request.html.erb]: There is a div that wraps the other action's views that will show up in a dialog (while the home page and the admin pages get full-width).

It's possible that we need to do something more clever with layouts, but in the meantime this page looks vastly different from the other pages.  With the addition of some wrapper divs and making the buttons 12 column we can get a more consistent display.

### Current
sponsor-form-before

### Modified
sponsor-form-after


```ruby
<div class='<%= dialog_column_class %>'>
  <h1 id="dialogTitle">Share with your proxy group?</h1>

  <%= render 'form_flashes' %>
  <%= render 'searchworks_item_information' %>
  <div class="<%= "#{label_column_offset_class} #{content_column_class}" %>">
    <%= bootstrap_form_for(current_request) do |f| %>
      <%= hash_to_hidden_fields(request_params) %>

      <h2><%= render_user_information %></h2>
      <%= f.form_group help: 'Anyone in your group can check the status of this request and pick it up when it’s available.' do %>
        <%= f.button 'Yes, share with my group.', class: 'btn btn-default col-md-12', value: 'true', name: 'proxy' %>
      <% end %>

      <%= f.form_group help: 'Only you can see and pick up this request.' do %>
        <%= f.button 'No, just me.', class: 'btn btn-cardinal col-md-12', value: 'false', name: 'proxy' %>
      <% end %>
    <% end %>
  </div>

</div>

```

Comment (cbeer) [inline app/views/requests/sponsor_request.html.erb]: Thanks, I've incorporated these changes.

File: app/views/requests/sponsor_request.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  <%= render 'form_flashes' %>
+  <%= render 'searchworks_item_information' %>
+
+  <div class="<%= "#{label_column_offset_class} #{content_column_class}" %>">
+    <%= bootstrap_form_for(current_request) do |f| %>
+      <%= hash_to_hidden_fields(request_params) %>
+
+      <h2><%= render_user_information %></h2>
+      <%= f.form_group help: 'Anyone in your group can check the status of this request and pick it up when it’s available.' do %>
+        <%= f.button 'Yes, share with my group.', class: 'btn btn-default', value: 'true', name: 'request[proxy]' %>
```
Comment (jkeck) [inline app/views/requests/sponsor_request.html.erb]: If you add `col-xs-12` to the button class these will behave like the similar buttons on the form.

**Comment by jkeck** [comment]
:+1: one minor comment, not a blocker.

**Comment by cbeer** [comment]
Sure, makes sense to me (although maybe we should refactor that into the CSS some time?)

File: app/views/requests/sponsor_request.html.erb (hunk 3 above)
Comment (cbeer) [inline app/views/requests/sponsor_request.html.erb]: Fixed in 92769b3.

**Comment by jkeck** [comment]
:+1: I think we currently have 3 button types.  Normal buttons, form buttons, and scan/page chooser buttons.  Maybe wait until they all shake out to figure out how to organize their styles.

**Pull request merged**